### PR TITLE
fix(influx publishing): the system id needs to be indexed

### PR DESCRIPTION
**Pull request by stakach**
otherwise we can't join the fields

### CODE REVIEW

**Comment by w-le** [comment]
hey @stakach will this still work for non-Logic drivers? iirc only Logic drivers are directl associated with a control system

**Comment by stakach** [comment]
yep

**Comment by stakach** [comment]
Still a work in progress @w-le - need to update specs and things before we merge this
If you can deploy it somewhere then we can test that these changes meet our needs

**Comment by w-le** [comment]
i don't really understand the code change itself so i will base my review on test results, after i deploy this feat branch to a live site.

I will attempt to write an influx query that shows all `Bookings` module states with their system name/id, over time.

I'll leave the code reviews to caspian and marvin

File: src/source/publishing/influx_publisher.cr [hunk 1]
```diff
@@ -71,6 +71,8 @@ module PlaceOS::Source
       tags = HIERARCHY.each_with_object({} of String => String) do |key, obj|
         obj["pos_#{key}"] = data.zone_mapping[key]? || "_"
       end
+      tags["pos_system"] = data.control_system_id
+      tags["pos_index"] = data.index.to_i64.to_s
```
Comment (caspiano) [inline src/source/publishing/influx_publisher.cr:75]: I don't think we transform existing points if there is a change in the index of a module.
That might a thing to consider when setting up systems with analytics.

File: src/source/publishing/influx_publisher.cr [hunk 2]
```diff
@@ -71,6 +71,8 @@ module PlaceOS::Source
       tags = HIERARCHY.each_with_object({} of String => String) do |key, obj|
```
Comment (caspiano) [inline src/source/publishing/influx_publisher.cr:71]: Save a tiny bit of memory.

```suggestion
      tags = HIERARCHY.each_with_object(Hash(String, String).new(initial_capapcity: HIERARCHY.size)) do |key, obj|
```

**Pull request merged**